Assets/Engine/Inventory/Item.cs

```diff
@@ -117,6 +147,61 @@ public virtual void Start()
             }
         }
 
+    //vvv collision noise code vvv
+    void OnCollisionEnter(Collision other) {
+        //Yeah, nested if statements, just trust me here bro.
+            //Only execute this code if we're supposed to make collision noises.
+            if(makesCollisionNoises){
+                //Check if there are any oddities with the configuration of the collision sounds.
+                if(!(lightImpactVelocity > hardImpactVelocity) && !(useHardImpactSounds && hardImpactSounds == null) && !(lightImpactSounds == null) && !(audioSource == null) && !(useBackupAudioSource && backupAudioSource == null))
+                {
+                //This section checks if we're using hard or light impact sounds, and assigns it to the best available audio source.
+                    if(useHardImpactSounds && other.relativeVelocity.magnitude > hardImpactVelocity){
+                        if(useBackupAudioSource && audioSource.isPlaying && !backupAudioSource.isPlaying){
+                            backupAudioSource.PlayOneShot(PickSound(hardImpactSounds), collisionVolume);
+                        }
+                        else if(!audioSource.isPlaying){
+                            audioSource.PlayOneShot(PickSound(hardImpactSounds), collisionVolume);
+                        }
+                    }
+                    else if(other.relativeVelocity.magnitude > lightImpactVelocity){
+                        if(useBackupAudioSource && audioSource.isPlaying && !backupAudioSource.isPlaying){
+                            backupAudioSource.PlayOneShot(PickSound(lightImpactSounds), collisionVolume);
+                        }
+                        else if(!audioSource.isPlaying){
+                            audioSource.PlayOneShot(PickSound(lightImpactSounds), collisionVolume);
+                        }
+                    }
+                }
+                else{
+                    //Logs a warning if there's any oddities in the configuration of collision sounds.
+                    Debug.LogWarning("<color=red>Woops!</color> " + gameObject.name + " is trying to play a collision sound but can't. Make sure the Item script's is configured correctly.");
+                }
+            }
+    }
+        
+    public AudioClip PickSound(AudioClip[] availableSounds){
+        //Pick a clip from the supplied array and return it
+        AudioClip currentClip = availableSounds[UnityEngine.Random.Range(0, availableSounds.Length)];
+        return currentClip;
+    }
+    private void GenerateAudioSource(){
+        audioSource = gameObject.AddComponent(typeof(AudioSource)) as AudioSource;
```

Comment:
I'm not sure about the behavior here. Does this add another audio source when one is already present?

If yes, check if `audioSource` is not `null` and return.

Reply:
Fixed!